### PR TITLE
Add support for generating a Oculus Mobile VR apk with hand tracking support

### PR DESCRIPTION
**Pull request by m4gr3d**
The PR also re-organizes the VR related features under their own section. They were previously under the `Graphics` section which didn't quite make sense:

- `graphics/xr_mode` -> `xr_features/xr_mode`
- `graphics/degrees_of_freedom` -> `xr_features/degrees_of_freedom`
- `xr_features/hand_tracking`

### CODE REVIEW

**Comment by m4gr3d** [comment]
cc @NeoSpark314

File: platform/android/export/export.cpp [hunk 1]
```diff
@@ -859,135 +859,175 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
...
 
-					int dof_index = p_preset->get("graphics/degrees_of_freedom"); // 0: none, 1: 3dof and 6dof, 2: 6dof
+					if (tname == "uses-feature") {
+						Vector<String> feature_names;
+						Vector<bool> feature_required_list;
+						Vector<int> feature_versions;
+
+						if (xr_mode_index == 1 /* XRMode.OVR */) {
+							// Check for degrees of freedom
+							int dof_index =
```
Comment (akien-mga) [inline platform/android/export/export.cpp]: The line break doesn't seem useful here, the broken line is just as long as the non broken one would be.

File: platform/android/export/export.cpp [hunk 2]
```diff
@@ -859,135 +859,175 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
...
+									p_preset->get("vr_features/degrees_of_freedom"); // 0: none, 1: 3dof and 6dof, 2: 6dof
+
+							if (dof_index > 0) {
+								feature_names.push_back("android.hardware.vr.headtracking");
+								feature_required_list.push_back(dof_index == 2);
+								feature_versions.push_back(1);
+							}
+
+							// Check for hand tracking
+							int hand_tracking_index = p_preset->get("vr_features/hand_tracking"); // // 0: none, 1: optional, 2: required
```
Comment (akien-mga) [inline platform/android/export/export.cpp]: Double `//`.

**Comment by akien-mga** [comment]
> Blocked on #34965.
> 
> The PR also re-organizes the VR related features under their own section. They were previously under the `Graphics` section which didn't quite make sense:
> 
>     * `graphics/degrees_of_freedom` -> `vr_features/degrees_of_freedom`
>     * `vr_features/hand_tracking`

Makes sense, but now it seems weird to have `xr_mode` still in `graphics` IMO, as it controls features of `vr_features` which are not necessarily graphical.

Maybe move all to `xr_features`?

**Comment by akien-mga** [comment]
Needs rebase after #34965.

**Comment by akien-mga** [comment]
I did not test yet but the code looks fine, and I think it's good to do those changes now without waiting for after the 3.2 release (especially since some export properties are renamed). Let's merge and get it tested in the next build :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**